ecs/pod_definition_manager.go

```diff
@@ -147,6 +147,24 @@ func (m *BasicPodDefinitionManager) CreatePodDefinition(ctx context.Context, opt
 	return &item, nil
 }
 
-func (m *BasicPodDefinitionManager) shouldCache() bool {
+// DeletePodDefinition deletes a pod definition and deletes it from the cache if
+// it is using a cache.
+func (m *BasicPodDefinitionManager) DeletePodDefinition(ctx context.Context, id string) error {
+	if _, err := m.client.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
```

Comment:
Any need to inspect the `DeregisterTaskDefinitionOutput` returned here and confirm it is non zero?

Reply:
The response only contains the original task definition that we registered (except it's set to inactive), which we don't need to use. I believe the only API calls that populate a `Failures` field in the response output are usually ones that can partly fail (like the bulk [`DescribeTasks`](https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#DescribeTasksOutput) operation where it might succeed in describing only some of the resources the user asks for). For this API call, it's an all-or-nothing kind of operation so it's assumed that if it failed to deregister, the error is populated.

Reply:
SGTM